### PR TITLE
Add script to verify issue 3580

### PR DESCRIPTION
**Pull request by ShobhitAd**
ATF Test Scripts to check #[3580](https://github.com/smartdevicelink/sdl_core/issues/3580)

This PR is **ready** for review.

### Summary
[Summary of PR changes]

### ATF version
develop

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: test_scripts/Defects/7_1/3580_Revoked_app_activated_via_device_consent.lua [hunk 1]
```diff
@@ -0,0 +1,64 @@
+---------------------------------------------------------------------------------------------------
+-- User story: https://github.com/SmartDeviceLink/sdl_core/issues/3580
+--
+-- Description:
+-- HMI responds with UNSUPPORTED_RESOURCE to Speak component of SubtleAlert
```
Comment (jacobkeeler) [inline test_scripts/Defects/7_1/3580_Revoked_app_activated_via_device_consent.lua:5]: Update description

**Review by jacobkeeler — CHANGES_REQUESTED** [review]

File: test_scripts/Defects/7_1/3580_Revoked_app_activated_via_device_consent.lua [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      local RequestId1 = common.hmi.getConnection():SendRequest("SDL.GetUserFriendlyMessage", {language = "EN-US", messageCodes = {"DataConsent"}})
+      local RequestId2 = common.hmi.getConnection():SendRequest("SDL.GetUserFriendlyMessage", {language = "EN-US", messageCodes = {"AppUnsupported"}})
+
+      EXPECT_HMIRESPONSE(RequestId1,{result = {code = 0, method = "SDL.GetUserFriendlyMessage"}})
+      :Do(function(_,_)
+        common.hmi.getConnection():SendNotification("SDL.OnAllowSDLFunctionality", {allowed = true, source = "GUI", device = {id = utils.getDeviceMAC(), name = utils.getDeviceName()}})
+        end)
+
+      EXPECT_HMIRESPONSE(RequestId2,{result = {code = 0, method = "SDL.GetUserFriendlyMessage"}})
+
```
Comment (jacobkeeler) [inline test_scripts/Defects/7_1/3580_Revoked_app_activated_via_device_consent.lua:49]: I think you would need to add another expectation here to match up with the issue (currently this test passes on develop):
```
      common.getHMIConnection():ExpectRequest("BasicCommunication.ActivateApp"):Times(0)
```

**Review by jacobkeeler — CHANGES_REQUESTED** [review]

File: test_scripts/Defects/7_1/3580_Revoked_app_activated_via_device_consent.lua (hunk 1 above)
Comment (ShobhitAd) [reply test_scripts/Defects/7_1/3580_Revoked_app_activated_via_device_consent.lua]: Added in https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/2505/commits/7ffb9e92e9d995f843068e807a2f61437c5db45b

File: test_scripts/Defects/7_1/3580_Revoked_app_activated_via_device_consent.lua (hunk 2 above)
Comment (ShobhitAd) [reply test_scripts/Defects/7_1/3580_Revoked_app_activated_via_device_consent.lua:69]: Added in https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/2505/commits/7ffb9e92e9d995f843068e807a2f61437c5db45b

**Review by jacobkeeler — APPROVED** [review]

**Pull request merged**